newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -0,0 +1,291 @@
+  );
+
+  // If the user has received this particular pack, either:
+  // - they just logged in, and already have it, so we close the dialog.
+  // - they just bought it, we display the success message.
+  React.useEffect(
+    () => {
+      if (loadedReceivedAssetPackInStore) {
+        const receivedAssetPack = loadedReceivedAssetPackInStore.find(
+          pack => pack.id === privateAssetPackListingData.id
+        );
+        if (receivedAssetPack) {
+          if (isBuying) {
+            setIsBuying(false);
+            setPurchaseSuccessful(true);
+          } else if (!purchaseSuccessful) {
+            onClose();
+          }
+        }
+      }
+    },
+    [
+      loadedReceivedAssetPackInStore,
+      privateAssetPackListingData,
+      isBuying,
+      onClose,
+      isCheckingPurchasesAfterLogin,
+      purchaseSuccessful,
+    ]
+  );
+
+  const dialogContents = !profile
+    ? {
+        subtitle: <Trans>Log-in to purchase this item</Trans>,
+        content: (
+          <CreateProfile
+            onLogin={onLogin}
+            onCreateAccount={onCreateAccount}
+            message={
+              <Trans>
+                Asset packs will be linked to your user account and available
+                for all your Projects. Log-in or Sign-up to purchase this pack.
+              </Trans>
+            }
+            justifyContent="center"
+          />
+        ),
+      }
+    : purchaseSuccessful
+    ? {
+        subtitle: <Trans>Your purchase has been processed!</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <Text>
+              <Trans>
+                You can close this window and go back to the asset store to
+                download your assets.
+              </Trans>
+            </Text>
+          </Line>
+        ),
+      }
+    : isBuying
+    ? {
+        subtitle: <Trans>Complete your payment on the web browser</Trans>,
+        content: (
+          <>
+            <Line justifyContent="center" alignItems="center">
+              <CircularProgress size={20} />
+              <Spacer />
+              <Text>Waiting for the purchase confirmation...</Text>
+            </Line>
+            <Spacer />
+            <Line justifyContent="center">
+              <BackgroundText>
+                <Trans>
+                  Once you're done, come back to GDevelop and the assets will be
+                  added to your account automatically.
+                </Trans>
+              </BackgroundText>
+            </Line>
+          </>
+        ),
+      }
+    : isCheckingPurchasesAfterLogin
+    ? {
+        subtitle: <Trans>Loading your profile...</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <CircularProgress size={20} />
+          </Line>
+        ),
+      }
+    : {
+        subtitle: (
+          <Trans>
+            The Asset Pack {privateAssetPackListingData.name} will be linked to
```

Comment:
in the first message, `asset pack` is written `Asset packs`, and here it's `Asset Pack`. I'm not used to use this kind of casing so I'm not sure what's the best to do about this inconsistency

Reply:
Bad habit I have, I think we should always have lower case